docs/basics/functional-programming-and-jotai.mdx

```diff
@@ -38,31 +38,97 @@ const greetingPromise = (async function () {
 })()
 ```
 
-This similarity is no coincidence. Both atoms and promises are **Monads**, a
-concept from **functional programming**. The syntax used in both examples is
-called **do-notation**, a syntax sugar for the plainer monad interface.
+This similarity is no coincidence. Both atoms and promises are **Monads**†, a
+concept from functional programming. The syntax used in both `greetingAtom` and
+`greetingPromise` is known as **do-notation**, a syntax sugar for the plainer
+monad interface.
 
----
+### About monads
 
 The monad interface is responsible for the fluidity of the atom and promise
 interfaces. The monad interface allowed us to define `greetingAtom` in terms of
 `nameAtom` and `countAtom`, and allowed us to define `greetingPromise` in terms
 of `namePromise` and `countPromise`.
 
-For the mathematically inclined, a structure (like `Atom` or `Promise`) is a
-monad if you can implement the following functions for it. A fun exercise is
-trying to implement `of`, `map` and `join` for Atoms, Promises and Arrays.
+If you're curious, a structure (like `Atom` or `Promise`) is a monad if you can
+implement the following functions for it. A fun exercise is trying to implement
+`of`, `map` and `join` for Arrays.
 
-```typescript
-type SomeMonad<T> = /* ... */
+```ts
+type SomeMonad<T> = /* for example... */ Array<T>
 declare function of<T>(plainValue: T): SomeMonad<T>
 declare function map<T, V>(
   anInstance: SomeMonad<T>,
-  transformContents: (contents: T) => V
+  transformContents: (contents: T) => V,
 ): SomeMonad<V>
 declare function join<T>(nestedInstances: SomeMonad<SomeMonad<T>>): SomeMonad<T>
 ```
 
+The shared heritage of Promises and Atoms means many patterns and best-practices
+can be reused between them. Let's take a look at one.
+
+### Sequencing
+
+When talking about callback hell, we often mention the boilerplate, the
+indentation and the easy-to-miss mistakes. However, plumbing a single async
+operation into another single async operation was not the end of the callback
+struggle. What if we made four network calls and needed to wait for them all?
+A snippet like this was common:
```

Comment:
Am I correct here? I did not code collaboratively before the ES6 Promise, so this is just from my experience.

Reply:
I'm not 100% sure if it's correct...